nicknamer.py

```diff
@@ -175,4 +180,64 @@ async def trace(context: Context) -> None:
         )
 
 
+@nicknamer.event
+async def on_message(message: Message) -> None:
+    cleaned_urls = {}
+
+    take_extreme_counter_measures = False
+
+    extractor = URLExtract()
+    for url in extractor.gen_urls(message.content):
+        clean_url = clear_url(url)
+
+        if url != clean_url:
+            cleaned_urls[url] = clean_url
+
+        take_extreme_counter_measures = take_extreme_counter_measures or len(
+            clean_url) * URL_LENGTH_VIOLATION_FACTOR < len(url)
+
+    if cleaned_urls:
```

Comment:
maybe we can early exit if this condition is not met - nothing to do, lowers indentation

Reply:
As far as I have been able to tell, there is no clear consensus on exit early vs only one return to clarify control flow. I was taught only one return unless it's totally unreadable to do so...

If you want to reduce the indentation a better way would just be to put the countermeasures into a function. I had imagined all of this will be refactored once we add another feature that involves processing every message, like shamebot. But, three indentation levels in my personal max, which is why I didn't do any of this now.